pkg/model/resource/resource.go

```diff
@@ -19,54 +19,113 @@ package resource
 import (
 	"fmt"
 	"strings"
-
-	"sigs.k8s.io/kubebuilder/v2/pkg/model/config"
 )
 
 // Resource contains the information required to scaffold files for a resource.
 type Resource struct {
-	// Group is the API Group. Does not contain the domain.
+	// Group is the resource's group. Does not contain the domain.
 	Group string `json:"group,omitempty"`
-
-	// GroupPackageName is the API Group cleaned to be used as the package name.
-	GroupPackageName string `json:"-"`
-
-	// Version is the API version.
+	// Domain is the resource's domain.
+	Domain string `json:"domain,omitempty"`
+	// Version is the resource's version.
 	Version string `json:"version,omitempty"`
-
-	// Kind is the API Kind.
+	// Kind is the resource's kind.
 	Kind string `json:"kind,omitempty"`
 
-	// Plural is the API Kind plural form.
+	// Plural is the resource's kind plural form.
 	Plural string `json:"plural,omitempty"`
 
-	// ImportAlias is a cleaned concatenation of Group and Version.
-	ImportAlias string `json:"-"`
+	// Path is the path to the go package where the types are defined.
+	Path string `json:"path,omitempty"`
 
-	// Package is the go package of the Resource.
-	Package string `json:"package,omitempty"`
+	// API holds the information related to the resource API.
+	API *API `json:"api,omitempty"`
 
-	// Domain is the Group + "." + Domain of the Resource.
-	Domain string `json:"domain,omitempty"`
+	// Controller specifies if a controller has been scaffolded.
+	Controller bool `json:"controller,omitempty"`
 
-	// Namespaced is true if the resource is namespaced.
-	Namespaced bool `json:"namespaced,omitempty"`
+	// Webhooks holds the information related to the associated webhooks.
+	Webhooks *Webhooks `json:"webhooks,omitempty"`
+}
+
+// QualifiedGroup returns the fully qualified group name with the available information.
+func (r Resource) QualifiedGroup() string {
+	switch "" {
+	case r.Domain:
+		return r.Group
+	case r.Group:
+		return r.Domain
+	default:
+		return fmt.Sprintf("%s.%s", r.Group, r.Domain)
+	}
+}
```

Comment:
It is not accurate. The Qualified Group also changes to work with core groups.See: https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/model/resource/options.go#L252-L259

Reply:
The `Domain` field is changed in case of core-groups, this method is valid for both.